mockgcp/mockserviceusage/serviceusagev1.go

```diff
@@ -273,6 +275,47 @@ func (s *ServiceUsageV1) ListServices(ctx context.Context, req *pb.ListServicesR
 		})
 	}
 
+	// Implement field filtering
+	// TODO: Can we mame this generic?  Is it used in other APIs?
+	{
+		md, _ := metadata.FromIncomingContext(ctx)
+		query := ""
+		if md != nil {
+			for _, v := range md.Get("query") {
+				query = v
+			}
+		}
+		fields := ""
+		if query != "" {
+			q, err := url.ParseQuery(query)
```

Comment:
Would it make sense to try to ParseQuery() before overriding an earlier query string?